inst/scripts/SEL3D.R

```diff
@@ -0,0 +1,251 @@
+#SEL3D.R
+#
+#' \code{SEL3D} wrapper function for getPdbIds. Blasts input sequence and
+#'              return MT object with match information.
+#'
+#' Details.
+#'
+#' @param targetSeq input sequence to be queried, must be cDNA of a length divisible by 3.
+#' @param convert bool to convert DNA sequences to amino acids, True by default
+#'                NOTE: The user should NOT change convert unless for testing purposes
+#' @param silent bool, set to True to turn off all messages.
+#' @return MT object - dataframe with headers queryid, subjectids, identity, positives,
+#'                     alignment length, mismatches, gapopens, q.start, q.end, s.start, s.end,
+#'                     evalue, bitscore, mlog.evalue, pdb.id
+#'
+#'
+#'
+#' @examples
+#' >SEL3D("AAGTCT")
+#' >sequence = "ACTGTGACTGTAAATGTT"
+#' >SEL3D(sequence)
+#' @export
+
+SEL3D <- function(targetSeq, convert = TRUE, silent = TRUE) {
+  #verify params
+  if(mode(convert) != "logical") {
+      stop(paste0("Mode error: argument requires logical input"))
+  }
+  if(mode(silent) != "logical") {
+      stop(paste0("Mode error: argument requires logical input"))
+  }
+  return(getPdbIds(targetSeq, convert = convert, silent = silent))
+}
+
+getPdbIds <- function(targetSeq, startPos = 1 , endPos = 1, convert = TRUE, silent = TRUE){
+    #getPdbIds
+    #
+    #' \code{getPdbIds} Blasts targetSeq to PDB and returns an MT object
+
+    #' Details.
+    #'
+    #' @param targetSeq input sequence to be queried, must be cDNA of a length divisible by 3.
+    #' @param startPos start position of the sequence, set to 1 by default, altered in recursive steps
+    #' @param endPos end position of sequence, set to 0, altered to aa length in recursive steps
+    #' @param convert bool to convert DNA sequences to amino acids, True by default
```

Comment:
We don't convert to amino acids on recursive calls because the sequence has already been converted.